File: 3-swagger-api-documentation/0-typescript/src/modules/dog/dog.service.ts
/**
 * Business logic service for Dog.
 */
import {
    Injectable
} from "@nestjs/common"

/** in-memory dog record.
 */
export interface DogRecord {
    name: string
    breed: string
}

/** demo dog business logic — list-only, sufficient for separate Swagger tag.
 */
@Injectable()
export class DogService {
    private dogs: DogRecord[] = [
        {
            name: "Rex",
            breed: "Labrador",
        },
    ]

    /** returns all demo dogs.
     * array of dog records.
     */
    findAll(): DogRecord[] {
        return this.dogs
    }
}

File: 3-swagger-api-documentation/backend/2-csharp/Dog/DogService.cs
namespace SwaggerDemo.Dog;

/// <summary>
/// in-memory dog record.
/// </summary>
/// <param name="Name">dog name.</param>
/// <param name="Breed">dog breed.</param>
public record DogRecord(string Name, string Breed);

/// <summary>
/// demo dog business logic — list-only, sufficient for a separate Swagger tag.
/// </summary>
public class DogService
{
    private readonly List<DogRecord> _dogs = new()
    {
        new DogRecord("Rex", "Labrador"),
    };

    /// <summary>
    /// returns all demo dogs.
    /// </summary>
    /// <returns>array of dog records.</returns>
    public IReadOnlyList<DogRecord> FindAll() => _dogs;
}
